src/plots/attributes.js

```diff
@@ -129,6 +129,19 @@ module.exports = {
         ].join(' ')
     },
     hoverlabel: fxAttrs.hoverlabel,
+    hovertemplate: {
```

Comment:
TODO: let's move this down to the trace `attributes.js` file.

By declaring `hovertemplate` per-trace, we'll be able to list of the (event data) keys available for each trace.

Reply:
Ha I see. You declared `hovertemplate` in `plots/attributes.js` but then overrode it in the trace modules. 

We should instead remove `hovertemplate` form `plots/attributes.js`, declare it in full in `scatter/attributes.js` and `extendFlat` its `description` in `bar`, `pie` and `histogram` attribute files.

Reply:
I made a function similar to `plots/domain.js`as we discussed but I still need to extend their description at the trace-level. Let me know if it looks good in 8e8b75b

Reply:
8e8b75b is brilliant. I'm a big fan of putting this new little module in `component/fx/` :ok_hand: